### ISSUE TITLE
[Bug]: Search failed with error `Channel: by-dev-rootcoord-dml_3_437061996750507423v1 returns err: err: failed to connect 10.102.8.29:21123, reason: context deadline exceeded` after querynode pod kill chaos test

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#20227]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20221031-a2d5947f
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2022-10-31T23:50:28.309Z] testcases/test_data_persistence.py::TestDataPersistence::test_milvus_default 

[2022-10-31T23:50:28.309Z] -------------------------------- live log setup --------------------------------

[2022-10-31T23:50:28.309Z] [2022-10-31 23:50:28 - INFO - ci_test]: ################################################################################ (conftest.py:189)

[2022-10-31T23:50:28.309Z] [2022-10-31 23:50:28 - INFO - ci_test]: [initialize_milvus] Log cleaned up, start testing... (conftest.py:190)

[2022-10-31T23:50:28.309Z] [2022-10-31 23:50:28 - INFO - ci_test]: [setup_class] Start setup class... (client_base.py:30)

[2022-10-31T23:50:28.309Z] [2022-10-31 23:50:28 - INFO - ci_test]: *********************************** setup *********************************** (client_base.py:36)

[2022-10-31T23:50:28.309Z] [2022-10-31 23:50:28 - INFO - ci_test]: [setup_method] Start setup test case test_milvus_default. (client_base.py:37)

[2022-10-31T23:50:28.563Z] -------------------------------- live log call ---------------------------------

[2022-10-31T23:50:28.563Z] [2022-10-31 23:50:28 - INFO - ci_test]: [test][2022-10-31T23:50:28Z] [0.00241571s] Hello_Milvus flush -> None (wrapper.py:30)

[2022-10-31T23:50:28.563Z] [2022-10-31 23:50:28 - INFO - ci_test]: assert create collection: 0.011652231216430664, init_entities: 6000 (test_data_persistence.py:28)

[2022-10-31T23:50:29.121Z] [2022-10-31 23:50:29 - INFO - ci_test]: [test][2022-10-31T23:50:28Z] [0.41753206s] Hello_Milvus insert -> (insert count: 3000, delete count: 0, upsert count: 0, timestamp: {self._timestamp}, success count: {self.succ_count}, err count: {self.err_count}) (wrapper.py:30)

[2022-10-31T23:50:29.121Z] [2022-10-31 23:50:29 - INFO - ci_test]: assert insert: 0.41777682304382324 (test_data_persistence.py:35)

[2022-10-31T23:50:32.373Z] [2022-10-31 23:50:32 - INFO - ci_test]: [test][2022-10-31T23:50:29Z] [3.03046782s] Hello_Milvus flush -> None (wrapper.py:30)

[2022-10-31T23:50:32.373Z] [2022-10-31 23:50:32 - INFO - ci_test]: [test][2022-10-31T23:50:32Z] [0.00245618s] Hello_Milvus flush -> None (wrapper.py:30)

[2022-10-31T23:50:32.373Z] [2022-10-31 23:50:32 - INFO - ci_test]: [test][2022-10-31T23:50:32Z] [0.00223936s] Hello_Milvus flush -> None (wrapper.py:30)

[2022-10-31T23:50:32.373Z] [2022-10-31 23:50:32 - INFO - ci_test]: assert flush: 3.0343682765960693, entities: 9000 (test_data_persistence.py:45)

[2022-10-31T23:50:32.373Z] [2022-10-31 23:50:32 - INFO - ci_test]: index info: [{'collection': 'Hello_Milvus', 'field': 'varchar', 'index_name': 'test_7epUTZyK', 'index_param': {'index_type': 'Trie'}}, {'collection': 'Hello_Milvus', 'field': 'float_vector', 'index_name': 'test_gf7kJEug', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}] (test_data_persistence.py:64)

[2022-10-31T23:50:32.373Z] [2022-10-31 23:50:32 - INFO - ci_test]: [test][2022-10-31T23:50:32Z] [0.00654403s] Hello_Milvus load -> None (wrapper.py:30)

[2022-10-31T23:50:37.601Z] [2022-10-31 23:50:37 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=Channel: by-dev-rootcoord-dml_3_437061996750507423v1 returns err: err: failed to connect 10.102.8.29:21123, reason: context deadline exceeded

[2022-10-31T23:50:37.601Z] , /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:277 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).Call

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/distributed/querynode/client/client.go:254 github.com/milvus-io/milvus/internal/distributed/querynode/client.(*Client).Search

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/proxy/task_search.go:484 github.com/milvus-io/milvus/internal/proxy.(*searchTask).searchShard

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/proxy/task_policies.go:127 github.com/milvus-io/milvus/internal/proxy.mergeRoundRobinPolicy.func1

[2022-10-31T23:50:37.601Z] /usr/local/go/src/runtime/asm_amd64.s:1571 runtime.goexit

[2022-10-31T23:50:37.601Z] Channel: by-dev-rootcoord-dml_2_437061996750507423v0 returns err: err: failed to connect 10.102.9.222:21123, reason: context deadline exceeded

[2022-10-31T23:50:37.601Z] , /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:277 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).Call

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/distributed/querynode/client/client.go:254 github.com/milvus-io/milvus/internal/distributed/querynode/client.(*Client).Search

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/proxy/task_search.go:484 github.com/milvus-io/milvus/internal/proxy.(*searchTask).searchShard

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/proxy/task_policies.go:127 github.com/milvus-io/milvus/internal/proxy.mergeRoundRobinPolicy.func1

[2022-10-31T23:50:37.601Z] /usr/local/go/src/runtime/asm_amd64.s:1571 runtime.goexit

[2022-10-31T23:50:37.601Z] )>, <Time:{'RPC start': '2022-10-31 23:50:32.088761', 'RPC error': '2022-10-31 23:50:37.092229'}> (decorators.py:108)

[2022-10-31T23:50:37.601Z] [2022-10-31 23:50:37 - ERROR - ci_test]: Traceback (most recent call last):

[2022-10-31T23:50:37.601Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 26, in inner_wrapper

[2022-10-31T23:50:37.601Z]     res = func(*args, **_kwargs)

[2022-10-31T23:50:37.601Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 57, in api_request

[2022-10-31T23:50:37.601Z]     return func(*arg, **kwargs)

[2022-10-31T23:50:37.601Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 719, in search

[2022-10-31T23:50:37.601Z]     res = conn.search(self._name, data, anns_field, param, limit, expr,

[2022-10-31T23:50:37.601Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 109, in handler

[2022-10-31T23:50:37.601Z]     raise e

[2022-10-31T23:50:37.601Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 105, in handler

[2022-10-31T23:50:37.601Z]     return func(*args, **kwargs)

[2022-10-31T23:50:37.601Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 136, in handler

[2022-10-31T23:50:37.601Z]     ret = func(self, *args, **kwargs)

[2022-10-31T23:50:37.601Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 85, in handler

[2022-10-31T23:50:37.601Z]     raise e

[2022-10-31T23:50:37.601Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 50, in handler

[2022-10-31T23:50:37.601Z]     return func(self, *args, **kwargs)

[2022-10-31T23:50:37.601Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 476, in search

[2022-10-31T23:50:37.601Z]     return self._execute_search_requests(requests, timeout, **_kwargs)

[2022-10-31T23:50:37.601Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 440, in _execute_search_requests

[2022-10-31T23:50:37.601Z]     raise pre_err

[2022-10-31T23:50:37.601Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 431, in _execute_search_requests

[2022-10-31T23:50:37.601Z]     raise MilvusException(response.status.error_code, response.status.reason)

[2022-10-31T23:50:37.601Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=Channel: by-dev-rootcoord-dml_3_437061996750507423v1 returns err: err: failed to connect 10.102.8.29:21123, reason: context deadline exceeded

[2022-10-31T23:50:37.601Z] , /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:277 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).Call

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/distributed/querynode/client/client.go:254 github.com/milvus-io/milvus/internal/distributed/querynode/client.(*Client).Search

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/proxy/task_search.go:484 github.com/milvus-io/milvus/internal/proxy.(*searchTask).searchShard

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/proxy/task_policies.go:127 github.com/milvus-io/milvus/internal/proxy.mergeRoundRobinPolicy.func1

[2022-10-31T23:50:37.601Z] /usr/local/go/src/runtime/asm_amd64.s:1571 runtime.goexit

[2022-10-31T23:50:37.601Z] Channel: by-dev-rootcoord-dml_2_437061996750507423v0 returns err: err: failed to connect 10.102.9.222:21123, reason: context deadline exceeded

[2022-10-31T23:50:37.601Z] , /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:277 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).Call

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/distributed/querynode/client/client.go:254 github.com/milvus-io/milvus/internal/distributed/querynode/client.(*Client).Search

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/proxy/task_search.go:484 github.com/milvus-io/milvus/internal/proxy.(*searchTask).searchShard

[2022-10-31T23:50:37.601Z] /go/src/github.com/milvus-io/milvus/internal/proxy/task_policies.go:127 github.com/milvus-io/milvus/internal/proxy.mergeRoundRobinPolicy.func1

[2022-10-31T23:50:37.602Z] /usr/local/go/src/runtime/asm_amd64.s:1571 runtime.goexit

[2022-10-31T23:50:37.602Z] )>

[2022-10-31T23:50:37.602Z]  (api_request.py:39)
```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test/detail/chaos-test/2636/pipeline

log:
[artifacts-querynode-pod-kill-2636-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/9906453/artifacts-querynode-pod-kill-2636-server-logs.tar.gz)

[artifacts-querynode-pod-kill-2636-pytest-logs.tar.gz](https://github.com/milvus-io/milvus/files/9906454/artifacts-querynode-pod-kill-2636-pytest-logs.tar.gz)

**Comment by zhuwenxing** [comment]
chaos type: pod-kill image tag: master-20221102-3ac87fe0 target pod: querynode
failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test/detail/chaos-test/2711/pipeline

[artifacts-querynode-pod-kill-2711-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/9925297/artifacts-querynode-pod-kill-2711-server-logs.tar.gz)
[artifacts-querynode-pod-kill-2711-pytest-logs.tar.gz](https://github.com/milvus-io/milvus/files/9925298/artifacts-querynode-pod-kill-2711-pytest-logs.tar.gz)

**Comment by yanliang567** [comment]
it reproduced on master-20221102-51acba7d, search on existing collections keeps failure since querynode pod kill
```
11/03/2022 02:13:49 AM - ERROR - <MilvusException: (code=1, message=fail to search on all shard leaders, err=Channel: yanliang-cluster-1cu-rootcoord-dml_3_437042918722571686v1 returns err: err: failed to connect 10.102.8.237:21123, reason: context deadline exceeded
, /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace
/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:277 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).Call
/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/client/client.go:254 github.com/milvus-io/milvus/internal/distributed/querynode/client.(*Client).Search
/go/src/github.com/milvus-io/milvus/internal/proxy/task_search.go:484 github.com/milvus-io/milvus/internal/proxy.(*searchTask).searchShard
/go/src/github.com/milvus-io/milvus/internal/proxy/task_policies.go:127 github.com/milvus-io/milvus/internal/proxy.mergeRoundRobinPolicy.func1
/usr/local/go/src/runtime/asm_amd64.s:1571 runtime.goexit
Channel: yanliang-cluster-1cu-rootcoord-dml_2_437042918722571686v0 returns err: err: failed to connect 10.102.8.237:21123, reason: context deadline exceeded
, /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace
/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:277 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).Call
/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/client/client.go:254 github.com/milvus-io/milvus/internal/distributed/querynode/client.(*Client).Search
/go/src/github.com/milvus-io/milvus/internal/proxy/task_search.go:484 github.com/milvus-io/milvus/internal/proxy.(*searchTask).searchShard
/go/src/github.com/milvus-io/milvus/internal/proxy/task_policies.go:127 github.com/milvus-io/milvus/internal/proxy.mergeRoundRobinPolicy.func1
/usr/local/go/src/runtime/asm_amd64.s:1571 runtime.goexit
```
milvus pod info:
```
yanliang-cluster-1cu-milvus-datanode-6f78dc5df4-jxrs2            1/1     Running     0              20h     10.102.6.237    devops-node10   <none>           <none>
yanliang-cluster-1cu-milvus-indexnode-667b679998-h5kh8           1/1     Running     0              20h     10.102.9.119    devops-node13   <none>           <none>
yanliang-cluster-1cu-milvus-mixcoord-5fcffb79b8-sqphq            1/1     Running     0              20h     10.102.5.102    devops-node21   <none>           <none>
yanliang-cluster-1cu-milvus-proxy-7cd485bd7f-ft4gg               1/1     Running     0              20h     10.102.9.118    devops-node13   <none>           <none>
yanliang-cluster-1cu-milvus-querynode-5cc6f48c54-bnh5k           1/1     Running     0              16h     10.102.10.80    devops-node20   <none>           <none>
```

**Comment by yah01** [comment]
/assign @zhuwenxing 
fixed with #20286

**Comment by zhuwenxing** [comment]
Not reproduced in `master-20221103-c08f1142`
So remove the critical label.
